internal/pkg/instrumentation/manager_test.go

```diff
@@ -186,3 +189,87 @@ func fakeManager(t *testing.T) *Manager {
 
 	return m
 }
+
+func TestRunStopping(t *testing.T) {
```

Comment:
Why not do this test for the `Instrumentation` object? That way the test will cover more code

Reply:
Yeah, that was the original plan. However, the `Instrumentation` type is in a different package and setting these testing overrides is not possible. We would need to create a "true" fake program to instrument or develop other testing override ability.

I'm planning to take a look at that in a follow-up, possibly. But didn't want to block this fix on that significant work.

Reply:
Tracking in https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/1006